pkg/utils/utils.go

```diff
@@ -277,26 +271,39 @@ func CustomizeUIRouteSpec(routeSpec *routev1.RouteSpec,
 // CustomizeDeployment ...
 func CustomizeDeployment(deploy *appsv1.Deployment, instance *kappnavv1.Kappnav) {
 	deploy.Labels = GetLabels(instance, deploy.Labels, &deploy.ObjectMeta, "")
+	
 	// Ensure that there's at least one replica
 	if deploy.Spec.Replicas == nil || *deploy.Spec.Replicas < 1 {
 		one := int32(1)
 		deploy.Spec.Replicas = &one
 	}
+	
+	// Add label selector 
 	deploy.Spec.Selector = &metav1.LabelSelector{
 		MatchLabels: map[string]string{
 			"app.kubernetes.io/component": deploy.GetName(),
 		},
 	}
+	
 }
 
 // CustomizePodSpec ...
 func CustomizePodSpec(pts *corev1.PodTemplateSpec, parentComponent *metav1.ObjectMeta,
 	containers []corev1.Container, volumes []corev1.Volume, instance *kappnavv1.Kappnav) {
+	
 	pts.Labels = GetLabels(instance, pts.Labels, parentComponent, "")
-	pts.Spec.Containers = containers
-	pts.Spec.RestartPolicy = corev1.RestartPolicyAlways
-	pts.Spec.ServiceAccountName = instance.GetName() + "-" + ServiceAccountNameSuffix
-	pts.Spec.Volumes = volumes
+	if pts.Spec.Containers == nil {
+		pts.Spec.Containers = containers
+	}
+	if pts.Spec.RestartPolicy == "" {
+		pts.Spec.RestartPolicy = corev1.RestartPolicyAlways
+	}
+	if pts.Spec.ServiceAccountName == "" {
+		pts.Spec.ServiceAccountName = instance.GetName() + "-" + ServiceAccountNameSuffix
```

Comment:
Similar comment as above. This value has to match the kAppNav service account name.